main.go

```diff
@@ -53,51 +60,54 @@ func main() {
 		panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
 	}
 	log := zapr.NewLogger(zapLog)
+	ctx := logr.NewContext(context.Background(), log)
 
-	err = run(log, args)
+	switch {
+	case args.Configuration != nil:
+		err = configurationCommand(ctx, args.Configuration)
+	case args.Registry != nil:
+		err = registryCommand(ctx, args.Registry)
+	}
```

Comment:
I would prefer having these in their own run() function and then returning in each case and then having a default that always returns an error.